internal/terraform/rootmodule/path.go

```diff
@@ -0,0 +1,12 @@
+package rootmodule
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+func pathEquals(path1, path2 string) bool {
+	volume1 := filepath.VolumeName(path1)
+	volume2 := filepath.VolumeName(path2)
+	return strings.EqualFold(volume1, volume2) && path1[len(volume1):] == path2[len(volume2):]
```

Comment:
I don't know enough about filesystems on Windows myself to decide, but I suspect we may need to compare of the other part of the path in case-insensitive way too at some point. It can be addressed as and when needed though.